generators/app/templates/src/screens/Dashboard/screens/Home/index.test.tsx

```diff
@@ -0,0 +1,72 @@
+import { render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import { actionCreators as authActions, ActionTypes, ResetUser } from 'contexts/UserContext/reducer';
+import { withContextProvider as withUserContext } from 'contexts/UserContext';
+import * as AuthService from 'services/AuthService';
+
+import { withContextProvider as withTechContext } from './context';
+
+import Home from '.';
+
+describe('Home component', () => {
+  afterAll(() => {
+    jest.clearAllMocks();
+  });
+
+  test('logouts when clicks logout button', async () => {
+    const logout = jest.fn();
+    const resetUser: () => ResetUser = jest.fn(() => ({ type: ActionTypes.RESET_USER }));
+    const removeCurrentUserToken = jest.fn();
+    jest.spyOn(AuthService, 'logout').mockImplementationOnce(logout);
+    jest.spyOn(authActions, 'resetUser').mockImplementationOnce(resetUser);
+    jest.spyOn(AuthService, 'removeCurrentUserToken').mockImplementationOnce(removeCurrentUserToken);
```

Comment:
I think all these mocks are signs of something being wrong.
Maybe `removeCurrentUserToken` should be called inside `logout` since both are part of the `AuthService`?

Also why do you need to know the implementation of `resetUser`? Can you just mock `const resetUser = jest.fn();`

Reply:
services are unit tested, because the component uses those services, they dont need to be tested again, that is why i dont care about the implementation, the component only needs to know that the function is actually called, and all of those services are called in the logout handle click. So the easiest way to retrieve coverage from it is to mock the implementation and check if it was actually called. Otherside, if we dont mock that services or actions, the code will work as the implementation were done. So we expect any graphic change to test but the logout action does not have any, the only thing is the redirection.